roles/tripleo_cluster_osds/tasks/zap.yml

```diff
@@ -0,0 +1,15 @@
+---
+- name: fail if device_list is empty
+  fail:
+    msg: "A device list should be passed as input of this role"
+  when: "{{ device_list | length == 0 }}"
+
+- name: Run 'ceph orch zap'
+  shell: "{{ ceph_cli }} orch device zap {{ item[0] }} {{ item[1] }} --force"
```

Comment:
note, this is relates to https://tracker.ceph.com/issues/45792

Reply:
> note, this is relates to https://tracker.ceph.com/issues/45792

Hi,
thanks for pointing out to the tracker! 
This definitely explains the issue on my env!
I like the `--all-available-devices` feature because on day2  operators shouldn't care about running commands/playbooks/scripts to add their new devices on the cluster.
Regarding the device zapping, since the scope of the project is  `focus on day1`, describe your cluster and push the button, this role will be removed or turned into something else (as per https://github.com/fmount/tripleo-ceph/issues/12#issuecomment-637086465 in the OpenStack context ironic cleaning should be sufficient) but I'm still using it for dev purposed until [1] is solved.

[1] https://tracker.ceph.com/issues/45792

Reply:
Thanks for the explanation!